### ISSUE TITLE
Feature Request: Automatically Tag Books with genres when scanned. 

### ISSUE THREAD

**Issue by TimHillier** [shockbytes/Dante#105]
I think it would be nice to have the books be tagged with their genres when they are scanned, that way you don't have to do it manually.

**Comment by shockbytes** [comment]
Hey Tim!

Thanks for this suggestion. I haven't thought about it. You know, I like to manually tag my books and like to have control over my own tags. But the more I think about it, the more I like it! I just need to verify if I get the genre information from the database (which I assume I'll get).

Cool, thanks! I'll put it in the backlog, if it's feasible.

**Comment by Shacamus** [comment]
FYI - I played a bit. 
Google JSON result contains a categories entry, which contains an array of category (string).

**Comment by shockbytes** [comment]
Cool @Shacamus!

Can you briefly outline the JSON result structure in this thread?

**Comment by Shacamus** [comment]
`{"title":"Red October","subtitle":"Left-Indigenous Struggles in Modern Bolivia","authors":["Jeffrey R. Webber"],"publisher":"BRILL","publishedDate":"2011-09-23","description":"In the opening years of this century, a left-indigenous insurrectionary cycle in Bolivia mounted the most radical challenge to neoliberalism in the Western hemisphere. This book provides a Marxist and indigenous-liberationist analysis of this revolutionary epoch and is historical context.","industryIdentifiers":[{"type":"ISBN_13","identifier":"9789004201552"},{"type":"ISBN_10","identifier":"9004201556"}],"readingModes":{"text":false,"image":true},"pageCount":376,"printType":"BOOK","categories":["History"],"maturityRating":"NOT_MATURE","allowAnonLogging":false,"contentVersion":"1.2.1.0.preview.1","imageLinks":{"smallThumbnail":"http://books.google.com/books/content?id=uQ4YlnD2cPAC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api","thumbnail":"http://books.google.com/books/content?id=uQ4YlnD2cPAC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"},"language":"en","previewLink":"http://books.google.es/books?id=uQ4YlnD2cPAC&pg=PA184&dq=red%2Boctober%2B&hl=&cd=1&source=gbs_api","infoLink":"http://books.google.es/books?id=uQ4YlnD2cPAC&dq=red%2Boctober%2B&hl=&source=gbs_api","canonicalVolumeLink":"https://books.google.com/books/about/Red_October.html?hl=&id=uQ4YlnD2cPAC"}`

>>"categories":["History"]


You can pick them using same code as for authors
`
return volumeInfo.get("categories")?.let { x ->
            x.asJsonArray
                    .joinToString(", ") { it.asString }
        } ?: ""
`

**Comment by shockbytes** [comment]
That's awesome @Shacamus!

Well, we could just pick those categories that are already created by the user. For example, if the user already created the label `History`, then the book will automatically attach the label to the book.

I'd avoid creating new labels for every category since the user might end up with a ton of unwanted labels.

**Comment by TimHillier** [comment]
What about giving the genre to the user as an option? So they still have control over if the books tagged by it or not, but then you wouldn't have to put in / or guess the genre? Or better yet, you could tell the user "hey this book is of genre "history" did you want to create/tag the genre?"

**Comment by shockbytes** [comment]
Yes, absolutely @TimHillier. It has to be optional since categories are always and English and probably won't work for non-English users. Otherwise, it seems quite feasible to allow users to automatically attach the genres to the book 👍 

I'll consider this feature for the upcoming releases. I can't make any promises when this will land in the app, since there are several other important features in the backlog.
